bin/specextract

```diff
@@ -1648,24 +1633,32 @@ def resps(args,specdicts,specextract_args,pars,parallelize):
     pardict_update_rmftool = parallelize(get_rmf_tool,args,progress=False)
     args = update_param_args_and_dict(pardict_update_rmftool,specdicts,specextract_args,pars)
 
-    asphists = {d["key_id"] : tempfile.NamedTemporaryFile(suffix=f"_asphist{d['chip_id']}",dir=d["tmpdir"]) for d in args}
-    asphists_name = {k:v.name for k,v in asphists.items()}
+    asphists = {d["key_id"] :
+                None if all([d["srcbkg"] == "bkg", not d["dobkgresp"], d["instrument"].upper() == "ACIS"])
+                else tempfile.NamedTemporaryFile(suffix=f"_asphist{d['chip_id']}",dir=d["tmpdir"])
+                for d in args}
+
+    tdetwmap = {d["key_id"] :
+                None if all([d["srcbkg"] == "bkg", not d["dobkgresp"]])
+                else tempfile.NamedTemporaryFile(suffix="_tdet",dir=d["tmpdir"])
+                for d in args}
 
-    tdetwmap = {d["key_id"] : tempfile.NamedTemporaryFile(suffix="_tdet",dir=d["tmpdir"]) for d in args}
-    tdetwmap_name = {k:v.name for k,v in tdetwmap.items()}
+    asphists_name = {k:v.name if v is not None else v for k,v in asphists.items()}
+    tdetwmap_name = {k:v.name if v is not None else v for k,v in tdetwmap.items()}
```

Comment:
I haven't followed the logic through below, so maybe the suggestion doesn't make sense, but can you drop those elements where `v is None` from the dictionaries here?

Reply:
I'll double check. I'm pretty sure that's what I originally tried, but things were going a bit wonky downstream in some regression tests without being so explicit here or I'd need to fall back on using some try/except statements.

Reply:
There cuold be a downstream reason why it's useful to keep the `None` values